### PR TITLE
[New Topic]D3.js: Base Pages

### PR DESCRIPTION
**Pull request by caupolicandiaz**
### Description

The edited files included are intended to initialize the D3 library topic in docs. It includes, the topic landing page as well as two concept pages that may serve as starting points for other concept and term entries to be developed.

### Type of Change

- Adding a new entry

### Checklist

- [/ ] All writings are my own.
- [/ ] My entry follows the Codecademy Docs style guide.
- [/ ] My changes generate no new warnings.
- [/ ] I have performed a self-review of my own writing and code.
- [/ ] I have checked my entry and corrected any misspellings.
- [/ ] I have made corresponding changes to the documentation if needed.
- [/ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [/ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [/ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/d3/concepts/anatomy/anatomy.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/computer-science'
+---
+
+The anatomy or structure of a D3 chart is built through the use of a range of methods to manipulate data and aesthetic parameters applied to an SVG object. A chart can be seen as meta-function, within it are objects and methods that generate the various components of a chart (e.g., paths, axes, labels, etc.) and at the end of it the result, or return, is an SVG that is the sum of all these intermediate operations.
+
+## Syntax
+
+In broad strokes, a chart definition will often take the following form.
```
Comment (dakshdeepHERE) [inline content/d3/concepts/anatomy/anatomy.md:20]:
```suggestion
In broad strokes, a chart definition will often take the following form:
```

File: content/d3/concepts/selection/selection.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+     ...
+```
+
+The pseudo code above represents a pattern commonly found in D3 for binding data and graphic elements, it is the `selectAll()/data()/join()/attr()` pattern. A selection begins with an initialization, `.selectAll("element")`, and then the elements that will populate this structure/allocation are defined in turn through the chained methods that follow.
+
+## Example
+
+A scatterplot is one of the simplest charts, it is a representation of data in 2-D space where each element is located based on an associated x-y coordinate value. Generally, each element is marked with a point, circle, or some other shape.
+
+For this example, the data will be consist of 50 points generated with the aid of the JavaScript `Math.random()` method. The set is constructed as follows.
```
Comment (dakshdeepHERE) [inline content/d3/concepts/selection/selection.md:37]:
```suggestion
For this example, the data will consist of 50 points generated with the aid of the JavaScript `Math.random()` method. The set is constructed as follows:
```

File: content/d3/concepts/selection/selection.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+     .attr("x", d => d[0])   // The element "selected" is now defined via attributes
+     .attr("y", d => d[1])
+     ...
+```
+
+The pseudo code above represents a pattern commonly found in D3 for binding data and graphic elements, it is the `selectAll()/data()/join()/attr()` pattern. A selection begins with an initialization, `.selectAll("element")`, and then the elements that will populate this structure/allocation are defined in turn through the chained methods that follow.
+
+## Example
+
+A scatterplot is one of the simplest charts, it is a representation of data in 2-D space where each element is located based on an associated x-y coordinate value. Generally, each element is marked with a point, circle, or some other shape.
```
Comment (dakshdeepHERE) [inline content/d3/concepts/selection/selection.md:35]:
```suggestion
A `scatterplot` is one of the simplest charts, it is a representation of data in 2-D space where each element is located based on an associated x-y coordinate value. Generally, each element is marked with a point, circle, or some other shape.
```

File: content/d3/concepts/selection/selection.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+   .join("circle") .             // The circle that will match with each data pt is
+     .attr("cx", d => x(d[0]))   // added and defined
+     .attr("cy", d => y(d[1]))
+     .attr("r", d => Math.random(1,10) * 30)
+     .attr("fill", "#f5b042")
+     .attr("opacity", .5);
+```
+
+In this selection the `.selectAll()` method is chained to the `svg` container, and the argument `"circle"` is given. Then:
+
```
Comment (dakshdeepHERE) [inline content/d3/concepts/selection/selection.md:69]:
```suggestion
```
We don't give a white space before the bullet points and its desc, right?

File: content/d3/concepts/selection/selection.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+Tags:
+  - 'Graphs'
+  - 'Libraries'
+  - 'Structure'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/computer-science'
+---
+
+Selection is the assignment and manipulation of elements in the construction of a chart. In D3 there are a range of methods that are used to add and filter elements in order to allocate specific data and apply modifications. The most widely used among these are the `.select()`, `.selectAll()` and `.append()` methods.
```
Comment (dakshdeepHERE) [inline content/d3/concepts/selection/selection.md:16]:
```suggestion
**`Selection`** is the assignment and manipulation of elements in the construction of a chart. In D3 there are a range of methods that are used to add and filter elements in order to allocate specific data and apply modifications. The most widely used among these are the `.select()`, `.selectAll()` and `.append()` methods.
```
The first occurrence of the entry method should be in bold, Right?

File: content/d3/concepts/anatomy/anatomy.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+Tags:
+  - 'Graphs'
+  - 'Libraries'
+  - 'Structure'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/computer-science'
+---
+
+The anatomy or structure of a D3 chart is built through the use of a range of methods to manipulate data and aesthetic parameters applied to an SVG object. A chart can be seen as meta-function, within it are objects and methods that generate the various components of a chart (e.g., paths, axes, labels, etc.) and at the end of it the result, or return, is an SVG that is the sum of all these intermediate operations.
```
Comment (dakshdeepHERE) [inline content/d3/concepts/anatomy/anatomy.md:16]:
```suggestion
The **`anatomy`** or structure of a D3 chart is built through the use of a range of methods to manipulate data and aesthetic parameters applied to an SVG object. A chart can be seen as a meta-function, within it are objects and methods that generate the various components of a chart (e.g., paths, axes, labels, etc.) and at the end of it the result, or return, is an SVG that is the sum of all these intermediate operations.
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Hey @caupolicandiaz, I have few changes from my side. Also, I'm eager to see your comment on some of my suggestions.

**Comment by caupolicandiaz** [comment]
@dakshdeepHERE Thanks for the review, I've added the changes.

**Review by dakshdeepHERE — APPROVED** [review]

**Review by SSwiniarski — APPROVED** [review]
@caupolicandiaz Looks good.

**Pull request merged**